### PR TITLE
Chore: Refactor Codebase

### PR DESCRIPTION
**Pull request by Akinmyde**
#### Description
Currently, our codebase has some discrepancies. It is important that we refactor our codebase for uniformity and readability.
#### Type of change

- [x] Refactoring

#### How Has This Been Tested?

- [x] Unit testing
- [x] End to End testing
#### Checklist:
- [x] change error messages format
- [x] seperate middlewares from validations
- [x] create new joi schemas
- [x] rename files to match conventions
- [x] add new files(middlewares, controllers, helpers)
#### PT-ID
N/A

#### Screenshots
N/A
#### Questions:
N/A

### CODE REVIEW

File: server/controllers/article.controllers.js [hunk 1]
```diff
@@ -1,130 +1,139 @@
...
+
+    const { userObj } = req.user;
+    req.body.user_id = userObj.id;
+
+    req.body = spaceTrimmer(req.body);
+    const article = await Article.create(req.body);
+
+    if (!req.body.is_draft) {
+      if (req.body.keywords) {
+        req.body.keywords.forEach(async keyword => {
```
Comment (rotimi-babalola) [inline server/controllers/article.controllers.js]: I'm still seeing the double if I talked about @sojida weren't you able to fix this?

Comment (sojida) [reply server/controllers/article.controllers.js]: @rotimi-babalola I intend to raise a PR to fix this when this code refactor PR is merged

File: server/controllers/reset-password.controllers.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      await notification.passwordReset(email);
+    }
+  } catch (error) {
+    return res.status(500).json({
+      errors: serverError(),
+    });
+  }
+};
+
+const acceptRequest = (req, res) => {
```
[the remark this answers is not in the document]
Comment (rotimi-babalola) [reply server/controllers/reset-password.controllers.js]: I think you should fix this complaint from HoundCI

File: server/helpers/search-database.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+const { User, Article } = model;
+
+const findUser = async email => {
+  try {
+    const user = await User.findOne({ where: { email } });
+    if (user) {
+      return user;
+    }
+    return null;
```
Comment (rotimi-babalola) [inline server/helpers/search-database.js]: Can you verify this but I think `findOne` returns null if nothing is found

File: server/helpers/search-database.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  }
+};
+
+const findArticle = async id => {
+  try {
+    const article = await Article.findOne({ where: { id } });
+    if (article) {
+      return article;
+    }
+    return null;
```
Comment (rotimi-babalola) [inline server/helpers/search-database.js]: Can you verify this but I think findOne returns `null` if nothing is found

File: server/helpers/server-error.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+const serverError = () => {
```
[the remark this answers is not in the document]
Comment (rotimi-babalola) [reply server/helpers/server-error.js]: Fix this error from HoundCI also

File: server/helpers/validations.js [hunk 6]
```diff
@@ -1,215 +1,113 @@
...
+  Object.keys(body).forEach(item => {
+    if (!editableProfileFields.includes(item)) {
+      valid = false;
     }
+  });
+
+  return valid;
+};
+
+const compareFieldWithToken = (field, token) => {
```
[the remark this answers is not in the document]
Comment (rotimi-babalola) [reply server/helpers/validations.js]: Fix this HoundCI warning

**Pull request merged**